### DISCUSSION TITLE
Rancher - Apply real certificate to ingress (Apache) is not working.

### DISCUSSION THREAD

**Discussion by thawatchaisap** [community/community#130504] [Programming Help]
### Select Topic Area

Question

### Body

I has created the Secret at **Rancher v2.7.8**  by menu **More Resources > Core > Secrets**. After I updated the real private key and certificate data from file. Then I added the **Secret** to my **Ingresses**  (at menu **Service Discovery > Ingresses**). My Ingresses is **httpd-service** (Apache) for my Web App. There're nothing occurs on my web site. It still show Not secure https://  on browser. What wrong with my step. Any recommend could you please advise, thank you so much.

**Comment by LikhithRanga** [comment]
Your Ingress YAML should have the `tls` section with the `secretName` referencing your Secret. Clear your browser cache or use incognito mode to bypass cached SSL/TLS information. Lastly, check the logs of your Ingress controller and Kubernetes events for any errors related to the Ingress resource.

**Reply by thawatchaisap** [reply]
Below is YAML file which have tls section reference to secretName. Which is not working. 
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  creationTimestamp: '2024-04-04T07:12:25Z'
  generation: 25
  managedFields:
    - apiVersion: networking.k8s.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:spec:
          f:defaultBackend:
            .: {}
            f:service:
              .: {}
              f:name: {}
              f:port:
                .: {}
                f:number: {}
          f:ingressClassName: {}
          f:rules: {}
          f:tls: {}
      manager: agent
      operation: Update
      time: '2024-04-04T07:12:25Z'
    - apiVersion: networking.k8s.io/v1
      fieldsType: FieldsV1
      fieldsV1:
        f:status:
          f:loadBalancer:
            f:ingress: {}
      manager: nginx-ingress-controller
      operation: Update
      subresource: status
      time: '2024-07-01T19:08:03Z'
  name: ingress-apache
  namespace: default
  resourceVersion: '35287440'
  uid: 6dfe977e-f485-4ece-b300-95ea1a369996
spec:
  defaultBackend:
    service:
      name: httpd-service
      port:
        number: 80
  ingressClassName: nginx
  rules:
    - host: wms.mywebsite.com
  tls:
    - hosts:
        - wms.mywebsite.com
      secretName: wms-secrets-tls
status:
  loadBalancer:
    ingress:
      - ip: 10.1.1.20
   ```

**Reply by LikhithRanga** [reply]
Firstly, ensure that the wms-secrets-tls secret in your Kubernetes cluster contains the correct private key and certificate data that match the domain wms.mywebsite.com. Check the logs of your Ingress controller (nginx-ingress-controller) for any errors related to loading or using the TLS secret. Additionally, review Kubernetes events (kubectl get events) for warnings or errors associated with your Ingress resource or the httpd-service pods. Sometimes, browsers cache SSL/TLS information aggressively, so try accessing your website in incognito mode or clear your browser cache to ensure you're seeing the current SSL/TLS status.

**Comment by ebndev** [comment]
Thanks for posting in the GitHub Community, @thawatchaisap! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion. Hopefully someone there might be able to give your a nudge in the right direction.
